types/transaction_types.go

```diff
@@ -1,169 +1,119 @@
 package types
 
-import "fmt"
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+type TransactionType uint64
 
-type TxType uint64
+// TxType is an alias for TransactionType
+// Deprecated: use TransactionType
+type TxType = TransactionType
 
 //Never change the order of these types.
 //If any types are deleted use a placeholder to prevent index number from changing.
 const (
-	TxTypeUnknown = TxType(iota)
-	TxTypeIdentityCreate
-	TxTypeTokenAccountCreate
-	TxTypeTokenTx
-	TxTypeDataChainCreate
-	TxTypeDataEntry //per 256 btes
-	TxTypeScratchChainCreate
-	TxTypeScratchEntry //per 256 bytes
-	TxTypeTokenCreate
-	TxTypeKeyUpdate //update keys on the keychain the identity
-	TxTypeMultisigTx
-	TxTypeStateQuery //sends a query to a chain and returns its state information
-	TxTypeCreateSigSpec
-	TxTypeCreateSigSpecGroup
-	TxTypeAddCredits
-	TxTypeUpdateKeyPage
+	TxUnknown = TransactionType(iota)
```

Comment:
Need more discussion about the needs for type changes.

If you want to freeze types, you should have explicit comments rather than use iota

Reply:
Explicit comments and iota are orthogonal. I can add comments to all of the types and still use iota.

IMO, using iota is easier to read and less error-prone than adding an explicit numeric value to each entry.

Reply:
All valid transaction types now have comments

Reply:
Iota is gone